glean-core/src/metrics/mod.rs

```diff
@@ -30,27 +34,44 @@ pub use self::string_list::StringListMetric;
 pub use self::time_unit::TimeUnit;
 pub use self::uuid::UuidMetric;
 
+/// The available metrics
+///
+/// This is the in-memory and persisted layout of a metric
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub enum Metric {
+    /// A boolean metric. See [`BooleanMetric`](struct.BooleanMetric.html) for more information.
     Boolean(bool),
+    /// A counter metric. See [`CounterMetric`](struct.CounterMetric.html) for more information.
     Counter(i32),
+    /// A datetime metric. See [`DatetimeMetric`](struct.DatetimeMetric.html) for more information.
     Datetime(DateTime<FixedOffset>, TimeUnit),
+    /// A string metric. See [`StringMetric`](struct.StringMetric.html) for more information.
     String(String),
+    /// A string list metric. See [`StringListMetric`](struct.StringListMetric.html) for more information.
     StringList(Vec<String>),
+    /// A UUID metric. See [`UuidMetric`](struct.UuidMetric.html) for more information.
     Uuid(String),
 }
 
+/// A `MetricType` describes common behavior across all metrics
 pub trait MetricType {
+    /// Access the stored metadata
     fn meta(&self) -> &CommonMetricData;
 
+    /// Access the stored metadata mutable
     fn meta_mut(&mut self) -> &mut CommonMetricData;
 
+    /// Whether this metric should currently be recorded
+    ///
+    /// This depends on the metrics own state, as determined by its metadata,
+    /// and whether upload is enabled on the Glean object.
     fn should_record(&self, glean: &Glean) -> bool {
         glean.is_upload_enabled() && self.meta().should_record()
     }
 }
 
 impl Metric {
+    /// The category the metric fits into
```

Comment:
Can you expand on what is this used for? (i.e. to decide where to put stuff in the ping?)